stages/org.osbuild.rhsm.facts

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+"""
+This stage allows storing artifact-properties of the Red Hat Subscription
+Manager (rhsm-facts) in the built image.
+"""
+
+import json
+import os
+import sys
+
+import osbuild.api
+
+
+SCHEMA = r"""
+"properties": {
+  "facts": {
+    "type": "array",
+    "description": "Array of Red Hat Subscription Manager facts.",
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "key": { "type": "string" },
+        "value": { "type": "string" }
+      }
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    path = os.path.join(tree, "usr/share/osbuild/self")
+    file = os.path.join(path, "rhsm.facts")
+
+    os.makedirs(path, exist_ok=True)
+
+    with open(file, "x", encoding="utf-8") as f:
+        json.dump({fact["key"]: fact["value"] for fact in options["facts"]}, f)
+
+    os.makedirs(os.path.join(tree, "etc/rhsm/facts"), exist_ok=True)
+
+    os.symlink(
+        file,
```

Comment:
This is not correct, is it? `file` is prefixed with `tree`, which should not leek into the symlink, right? My previous comment was a bit confusing, probably. I explicitly dropped the `tree` prefix from my suggestion when sharing the path.

Reply:
It's not correct, I got overzealous.